app/src/main/java/com/evanfuhr/pokemondatabase/activities/MainActivity.java

```diff
@@ -133,8 +147,76 @@ public boolean onOptionsItemSelected(MenuItem item) {
         return true;
     }
 
-    void onClickMenuSetGame(MenuItem item) {
-        Toast toast = Toast.makeText(getApplicationContext(), MENU_ITEM_NOT_IMPLEMENTED_YET, Toast.LENGTH_LONG);
-        toast.show();
+    public void onClickMenuSetGame() {
+
+        // Get list of versions
+        VersionDAO versionDAO = new VersionDAO(this);
+        List<Version> versionList = versionDAO.getAllVersions();
+
+        // Get view
+        LayoutInflater layoutInflater = LayoutInflater.from(MainActivity.this);
+        View setGameVersionView = layoutInflater.inflate(R.layout.dialog_set_game_version, null);
+
+        // Setup spinner
+        Spinner versionSpinner = (Spinner) setGameVersionView.findViewById(R.id.spinner_game_version);
+        final VersionSpinnerAdapter versionSpinnerAdapter = new VersionSpinnerAdapter(this, R.layout.dialog_set_game_version, versionList);
+        versionSpinner.setAdapter(versionSpinnerAdapter);
+        // TODO: Set position
+        versionSpinner.setSelection(versionSpinnerAdapter.getPositionByVersion(_version_id));
+
+        versionSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+                Version version = versionSpinnerAdapter.getItem(position);
+                _version_id = version.getID();
+
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> parent) {
+
+            }
+        });
+
+        // Setup a dialog window
+        AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(MainActivity.this);
+        alertDialogBuilder.setTitle("Set Game Version");
+        alertDialogBuilder.setView(setGameVersionView);
+        alertDialogBuilder.setCancelable(true)
+                .setPositiveButton("Save", new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface dialog, int id) {
+                        // Save game version
+                        dialog.dismiss();
+                    }
+                })
+                .setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface dialog, int id) {
+                        restorePreferences();
+                        dialog.dismiss();
+                    }
+                });
+
+
+        AlertDialog alert = alertDialogBuilder.create();
+        alert.show();
+    }
+
+    private void restorePreferences() {
+        SharedPreferences settings = getSharedPreferences(String.valueOf(R.string.gameVersionID), MODE_PRIVATE);
+        _version_id = settings.getInt(String.valueOf(R.string.gameVersionID), 28); // Default game is Moon
```

Comment:
Where is the `28` coming from? Can we set this as a constant with the other values at the top of this file or some centralized constants file?

Reply:
Yes

Reply:
Done